yb_migrate/src/utils/diagnostics.go

```diff
@@ -0,0 +1,143 @@
+	TotalSize             int64     `json:"total_size"`              //done
+	LargestTableRows      int64     `json:"largest_table_rows"`      //done
+	LargestTableSize      int64     `json:"largest_table_size"`      //done
+	TargetClusterLocation string    `json:"target_cluster_location"` //ask on slack channels for this (answer given is insufficient)
+	TargetDBCores         int       `json:"target_db_cores"`         //unknown for now
+	SourceCloudDBType     string    `json:"source_cloud_type"`       //unknown for now
+}
+
+//Fill in primary-key based fields, if needed
+func InitJSON(exportdir string) {
+	var err error
+	jsonFilePath = filepath.Join(exportdir, "metainfo", "diagnostics.json")
+	_, err = os.OpenFile(jsonFilePath, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		ErrExit("Error while creating/opening diagnostics.json file: %v", err)
+	}
+	jsonBuf, err = os.ReadFile(jsonFilePath)
+	if err != nil {
+		ErrExit("Error while reading diagnostics.json file: %v", err)
+	}
+
+	if len(jsonBuf) != 0 {
+		err = json.Unmarshal(jsonBuf, &sendReq)
+		if err != nil {
+			ErrExit("Invalid diagnostics.json file: %v", err)
+		}
+	}
+
+	if sendReq.MigrationUuid == uuid.Nil {
+		sendReq.MigrationUuid, err = uuid.NewUUID()
+		sendReq.StartTime = time.Now().Format("2006-01-02 15:04:05")
+		if err != nil {
+			ErrExit("Error while generating new UUID for diagnostics.json: %v", err)
+		}
+	}
+
+}
+
+//Getter method for updating payload
+func GetPayload() *sendRequest {
+	return &sendReq
+}
+
+//Save payload to diagnostics.json
+func PackPayload(exportdir string) {
+	jsonBuf, _ = json.Marshal(sendReq)
+	os.WriteFile(jsonFilePath, jsonBuf, 0644)
+
+}
+
+//Send http request to flask servers
+func SendPayload() {
+	if !SendDiagnostics {
+	} else {
+
+		sendReq.LastUpdatedTime = time.Now().Format("2006-01-02 15:04:05")
+		postBody, _ := json.Marshal(sendReq)
+		requestBody := bytes.NewBuffer(postBody)
+
+		log.Infof("Payload being sent for diagnostic usage: %s\n", string(postBody))
+		resp, err := http.Post("http://127.0.0.1:8000/", "application/json", requestBody)
+
+		if err != nil {
+			ErrExit("Error while sending diagnostic data: %v", err)
+		}
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			ErrExit("Error while reading HTTP response from call-home server: %v", err)
+		}
+		log.Infof("HTTP response after sending diagnostic.json: %s\n", string(body))
+	}
+}
+
+func UpdateDataSize(exportdir string) {
+	datadirfiles := filepath.Join(exportdir, "data", "*_data*")
+	// totalSizeCmd := exec.Command("du", "-s", datadir)
+	// stdout, err := totalSizeCmd.CombinedOutput()
+	// if err != nil {
+	// 	ErrExit("Error while executing command for diagnostics data: %s\n%v", totalSizeCmd, err)
+	// }
+	// totalSize := strings.Split(string(stdout), "\t")[0]
+
+	files, err := filepath.Glob(datadirfiles)
+	if err != nil {
+		ErrExit("Error while matching files in data dir for diagnostics: %v", err)
+	}
+	var fileInfo fs.FileInfo
```

Comment:
why declare beforehand? Just use `:=` wherever possible, which will reduce the extra lines.

Reply:
done